src/server/common/helpers/redis-helper.js

```diff
@@ -29,25 +29,27 @@ class RedisHelper {
     return await this.client.set(fileId, JSON.stringify(fileDetails))
   }
 
-  async findUploadWithFiles(uploadId) {
-    const result = await this.findUploadDetails(uploadId)
-    if (!result || !result.fileIds) {
-      return result
-    }
-    const files = {}
-    for (const fileId of result.fileIds) {
-      const file = await this.findFileDetails(fileId)
-      if (file) {
-        files[fileId] = file
-        this.updateField(result.fields, fileId, {
-          s3Key: file?.s3Key,
-          s3Bucket: file?.s3Bucket,
-          fileStatus: file?.fileStatus
-        })
+  async findUploadAndFiles(uploadId) {
+    const uploadDetails = await this.findUploadDetails(uploadId)
+    const files = []
+
+    if (uploadDetails && uploadDetails.fileIds) {
+      for (const fileId of uploadDetails.fileIds) {
```

Comment:
I wonder if building up the array of promises and using `Promise.all` here would be more efficient?. Or maybe even using `Promise.allSettled()` - https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/allSettled